File: Luogu/JCA/JC1/P_1098_NOIP_2007_提高组_字符串的展开.cpp
#include <bits/stdc++.h>
using namespace std;
#define endl '\n'

int main() {
    ios::sync_with_stdio(false); cin.tie(nullptr); cout.tie(nullptr);
    int p1, p2, p3;
    string s;
    cin >> p1 >> p2 >> p3 >> s;
    int n = s.size();
    string ans = "";
    for (int i = 0; i < n; i++) {
        char ch = s[i];
        char pre = (i > 0) ? ans.back() : '#';
        char suf = (i + 1 < n) ? s[i + 1] : '#';
        if (ch == '-' && pre < suf && (islower(pre) && islower(suf) || isdigit(pre) && isdigit(suf))) {
            string tmp = "";
            if (p1 == 1) {
                for (char k = pre + 1; k < suf; k++) {
                    tmp += string(p2, k);
                }
            } else if (p1 == 2) {
                for (char k = pre + 1; k < suf; k++) {
                    tmp += string(p2, toupper(k));
                }
            } else {
                tmp += string((suf - pre - 1) * p2, '*');
            }
            if (p3 == 1) {
                ans += tmp;
            } else {
                ans += string(tmp.rbegin(), tmp.rend());
            }
        } else {
            ans += ch;
        }
    }
    cout << ans << endl;
    return 0;
}

File: Luogu/Course/JCA/JC1/P_1098_NOIP_2007_提高组_字符串的展开.py
p1, p2, p3 = map(int, input().split())
s = input()

n = len(s)
ans = ""

for i, ch in enumerate(s):
    pre = ans[-1] if i > 0 else "#"
    suf = s[i + 1] if i + 1 < n else "#"
    if ch == '-' and pre < suf and (pre.islower() and suf.islower() or pre.isdigit() and suf.isdigit()):
        tmp = ""
        if p1 == 1:
            for k in range(ord(pre) + 1, ord(suf)):
                tmp += chr(k) * p2
        elif p1 == 2:
            for k in range(ord(pre) + 1, ord(suf)):
                tmp += chr(k).upper() * p2
        else:
            tmp += '*' * (ord(suf) - ord(pre) - 1) * p2
        if p3 == 1:
            ans += tmp
        else:
            ans += tmp[::-1]
    else:
        ans += ch
print(ans)